### DISCUSSION TITLE
How to change performance folders from a midi keyboard

### DISCUSSION THREAD

**Discussion by Alan-Angel** [probonopd/MiniDexed#770] [General]
Hi, I'm pulling my hair out a bit. My keyboard can send MSB and LSB messages, I just want to go from the initial 127 performances to the ones in the folder 002 Greenhouse (and a user folder for me ideally). Can it be done? I run without the encoder/screen.  Thanks.

**Comment by soyersoyer** [comment]
Have you tried setting PerformanceSelectChannel to omni mode?

**Reply by Alan-Angel** [reply]
Hi, I set it to17, does that make it omni in this case?

**Comment by diyelectromusic** [comment]
What messages is your keyboard sending?  Performances are arranged in banks, so you need to send BANKSEL messages and then ensure you've configured MiniDexed to direct them to performances using PerformanceSelectChannel.

From https://github.com/probonopd/MiniDexed/wiki/Files#midi

_"PerformanceSelectChannel determines if MIDI Program Change (PC) messages should select individual voices within a tone generator or act at the system level and select performances. This mirrors the similarly named control on a DX7II. Setting this to 0 (the default) means PC messages select voices on any tone generators set to the receiving MIDI channel. Setting this to a MIDI channel (1 to 16) means PC messages on that channel will select performances. Setting this to a value >16 selects "Omni" mode so any PC messages received select performances. Note when in performance mode, no PC messages will be passed on to tone generators. Note that MIDIRXProgramChange has to be set to 1 for this to have any affect."_

I'll update the wiki to make it clear this affects both bank and program selection.

Kevin

**Reply by Alan-Angel** [reply]
Hi I'm pretty sure I'm in omni mode. The keyboard (M-Audio Oxygen MK IV) is a bit of a mess. You can send an LSB value, or an MSB value. Or a combined message that sends Program Change, LSB, and MSB. But then again the manual says you have to send program change after a bank change for it to change the bank which makes no sense. There's also a long list of commands in the manual, one of which being 'bank select 00' but there are no examples of how to use it. If anyone can tell me what message I should be sending minidexed to switch banks/folders maybe I can work back from there.

**Reply by diyelectromusic** [reply]
I think they often recommend BANKSEL then PC because it can be undefined which program to select after a bank change - should a device switch to program 1, switch to the same program number you were on in the new bank, or just change the bank with no voice change?

Actually the MIDI Spec does actually say: "The transmitter must transmit the MSB and LSB as a pair, and the Program Change must be sent immediately after the Bank Select pair."

MiniDexed expects the order: MSB then LSB and only acts on the change when it receives the LSB (using whatever MSB was last stored).  So you could probably get away with just sending the LSB if you only have less than 128 banks.

But IIRC we don't actually change the voice when receiving a BANKSEL - it is just the "new active bank" (you should see it change in the menus if you have Bank selected) but it will not load up a new voice from that bank until a voice is selected either via the menu or with a PC message.

So yes, you need the order: CC BANKSEL MSB (kind of optional if <128 banks), CC BANKSEL LSB, PC to actually hear a difference.

Kevin
